sqlglot/dialects/duckdb.py

```diff
@@ -268,6 +268,114 @@ def _json_extract_value_array_sql(
     return self.sql(exp.cast(json_extract, to=exp.DataType.build(data_type)))
 
 
+def _escape_regex_metachars(
+    self: DuckDB.Generator, delimiters: t.Optional[exp.Expression], delimiters_sql: str
+) -> str:
+    if not delimiters:
+        return delimiters_sql
+
+    REGEX_LITERAL_ESCAPES = {
+        "\\": "\\\\",  # literals need two slashes inside []
+        "-": "\\-",
+        "^": "\\^",
+        "[": "\\[",
+        "]": "\\]",
+    }
+
+    if isinstance(delimiters, exp.Literal) and delimiters.is_string:
+        literal_value = delimiters.this
+        escaped_literal = "".join(REGEX_LITERAL_ESCAPES.get(ch, ch) for ch in literal_value)
+        return self.sql(exp.Literal.string(escaped_literal))
+
+    REGEX_ESCAPE_REPLACEMENTS = (
+        ("\\", "\\\\"),
+        ("-", r"\-"),
+        ("^", r"\^"),
+        ("[", r"\["),
+        ("]", r"\]"),
+    )
```

Comment:
How is this different from `REGEX_LITERAL_ESCAPES`? Seems like the same mapping.